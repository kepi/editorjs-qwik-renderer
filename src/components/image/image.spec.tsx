import { createDOM } from "@builder.io/qwik/testing"
import { describe, expect, it } from "vitest"
import Image, { type ImageBlockData } from "./image"

describe("<Image />", async () => {
  describe("when receives a Image block with actions", async () => {
    const data: ImageBlockData = {
      file: {
        url: "https://cdn.directions.pt/uploads/2020/08/681-2000x1300-2.jpg",
        name: "681-2000x1300-2",
      },
      caption: "Deep in the <b>universe</b>",
      withBorder: true,
      stretched: true,
      withBackground: true,
    }

    it("renders a <figure> block with <img> and <figcaption>", async () => {
      const { screen, render } = await createDOM()
      await render(<Image data={data} class="special-image" />)
      expect(screen.innerHTML).toMatchSnapshot()
    })
  })

  describe("when receives a Image block without actions", async () => {
    const data: ImageBlockData = {
      file: {
        url: "https://cdn.directions.pt/uploads/2020/08/681-2000x1300-2.jpg",
        name: "681-2000x1300-2",
      },
      caption: "Deep in the universe",
      withBorder: false,
      stretched: false,
      withBackground: false,
    }

    it("renders a <figure> block with <img> and <figcaption>", async () => {
      const { screen, render } = await createDOM()
      await render(<Image data={data} />)
      expect(screen.innerHTML).toMatchSnapshot()
    })
  })

  describe("when receives a Image block without name", async () => {
    const data: ImageBlockData = {
      file: {
        url: "https://cdn.directions.pt/uploads/2020/08/681-2000x1300-2.jpg",
      },
      caption: "Deep in the universe",
      withBorder: false,
      stretched: false,
      withBackground: false,
    }

    it("renders a <figure> block with <img> and <figcaption>", async () => {
      const { screen, render } = await createDOM()
      await render(<Image data={data} />)
      expect(screen.innerHTML).toMatchSnapshot()
    })
  })

  describe("when receives a SimpleImage block", async () => {
    const data: ImageBlockData = {
      url: "https://cdn.directions.pt/uploads/2020/08/681-2000x1300-2.jpg",
      caption: "Deep in the universe",
      withBorder: false,
      stretched: false,
      withBackground: false,
    }

    it("renders a <figure> block with <img> and <figcaption>", async () => {
      const { screen, render } = await createDOM()
      await render(<Image data={data} />)
      expect(screen.innerHTML).toMatchSnapshot()
    })
  })
})
