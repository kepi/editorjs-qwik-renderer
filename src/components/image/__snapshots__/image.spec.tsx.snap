// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Image /> > when receives a Image block with actions > renders a <figure> block with <img> and <figcaption> 1`] = `"<!--qv --><figure class=\\"special-image image-block--stretched image-block--with-border image-block--with-background\\"><img alt=\\"Deep in the <b>universe</b>\\" src=\\"https://cdn.directions.pt/uploads/2020/08/681-2000x1300-2.jpg\\"><figcaption>Deep in the <b>universe</b></figcaption></figure><!--/qv-->"`;

exports[`<Image /> > when receives a Image block without actions > renders a <figure> block with <img> and <figcaption> 1`] = `"<!--qv --><figure><img alt=\\"Deep in the universe\\" src=\\"https://cdn.directions.pt/uploads/2020/08/681-2000x1300-2.jpg\\"><figcaption>Deep in the universe</figcaption></figure><!--/qv-->"`;

exports[`<Image /> > when receives a Image block without name > renders a <figure> block with <img> and <figcaption> 1`] = `"<!--qv --><figure><img alt=\\"Deep in the universe\\" src=\\"https://cdn.directions.pt/uploads/2020/08/681-2000x1300-2.jpg\\"><figcaption>Deep in the universe</figcaption></figure><!--/qv-->"`;

exports[`<Image /> > when receives a SimpleImage block > renders a <figure> block with <img> and <figcaption> 1`] = `"<!--qv --><figure><img alt=\\"Deep in the universe\\" src=\\"https://cdn.directions.pt/uploads/2020/08/681-2000x1300-2.jpg\\"><figcaption>Deep in the universe</figcaption></figure><!--/qv-->"`;
